python/tank_vendor/shotgun_authentication/console_authentication.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Console based authentication.
+"""
+
+from . import session_cache
+from .errors import AuthenticationError, AuthenticationCancelled
+
+from getpass import getpass
+import logging
+
+logger = logging.getLogger("shotgun_authentication").getChild(
```

Comment:
is it easier to say `logging.getLogger("shotgun_auth.console")`? Also suggest saying `console` rather than `interactive_authentication`.